test/unit/saveArticles.test.ts

```diff
@@ -238,3 +238,26 @@ test('--customFlavour', async (t) => {
     t.ok(ParisDocument.querySelector('#PRE_PROCESSOR'), `Paris was correctly pre-processed`);
     t.ok(PragueDocument.querySelector('#POST_PROCESSOR'), `Prague was correctly post-processed`);
 });
+
+test('treat one subtitle', async(t) => {
+    const { downloader, mw, dump } = await setupScrapeClasses({ format: '' });
+
+    // Wikicode is taken from article "Mechanical energy" which has a video with subtitle
+    const wikicode = `[[File:Physicsworks.ogv|thumb|200px|alt="Lecture demonstrating conservation of mechanical energy"|MIT professor [[Walter Lewin]] demonstrating conservation of mechanical energy]]`;
+    const htmlDoc = domino.createDocument((await convertWikicodeToHtml(wikicode, dump.mwMetaData.base)).data);
+    const contentRes = await treatSubtitles(htmlDoc.querySelector('track'), 'en.wikipedia.org', mw, 'Mechanical energy');
+
+    t.ok(contentRes, 'Video with subtitle rewriting');
+});
+
+test('treat multiple subtitles in one video', async(t) => {
+    const { downloader, mw, dump } = await setupScrapeClasses({ format: '' });
+
+    // Wikicode is taken from article "User:Charliechlorine/sandbox" which has multiple(4) subtitles in this video
```

Comment:
Why you don't use `testHtmlRewritingE2e` here (and in other places)?

Reply:
Are you indicating to replace `convertWikicodeToHtml` to `testHtmlRewritingE2e`. It is bit unclear to me what exactly needs to be done here?

Reply:
Replace 'treat one subtitle' and 'treat multiple subtitles in one video' ... or add a new test.... which is only based on `testHtmlRewritingE2e` and test that the HTML for a video with multiple subtitles is OK.

Reply:
My last comment is confusing:
* 'treat one subtitle' should test exactly which URL
* 'treat multiple subtitles in one video' should do the same
* I would add an global test, testing the whole HTML testHtmlRewritingE2e